drivers/include/soft_spi.h

```diff
@@ -13,10 +13,25 @@
  *
  * This module provides a software implemented Serial Peripheral Interface bus.
  * It is intended to be used in situation where hardware spi is not available.
- * The signatures of the functions are similar to the functions declared in spi.h
- * The clock speed is approximated by using xtimer_nanosleep.
- * Currently only the use of MOSI in master mode is implemented. Therefore receiving
- * data from a slave is currently not possible.
+ * The signatures of the functions are identical to the functions declared in
+ * `spi.h`. The clock speed is approximated by using `xtimer_nanosleep()`.
```

Comment:
@maribu is it worth using nanosleep here? AFAIK most xtimer backends are using a < 1MHz timer anyway, so best result achievable would be 1us no? `ztimer` could use `ptp` but that is another story.

Reply:
I very much agree.

Note that I only changed the wording of the doc in this PR, the use of `xtimer_nanosleep()` is a preexisting design decision. Maybe it is best to tackle this in a separate PR?

Reply:
I agree, can you rebase this one?

Reply:
I rebased it on master. I will later retest again, the rebase was a bit involved and I hope I didn't mess anything up.